functions/New-DbaComputerCertificate.ps1

```diff
@@ -0,0 +1,294 @@
+
+.PARAMETER Silent 
+Use this switch to disable any kind of verbose messages
+
+.NOTES
+Tags: Certificate
+
+Website: https://dbatools.io
+Copyright: (C) Chrissy LeMaire, clemaire@gmail.com
+License: GNU GPL v3 https://opensource.org/licenses/GPL-3.0
+
+.EXAMPLE
+New-DbaCertificate
+Creates a computer certificate for the local machine with the keylength of 1024.
+
+.EXAMPLE
+New-DbaCertificate -ComputerName Server1
+
+Creates a computer certificate _on the local machine_ for server1 with the keylength of 1024. 
+	
+The certificate is then copied to the new machine over WinRM and imported.
+
+.EXAMPLE
+New-DbaCertificate -ComputerName Server1 -WhatIf
+
+Shows what would happen if the command were run
+
+#>
+	[CmdletBinding(SupportsShouldProcess = $true, ConfirmImpact = "Low")]
+	param (
+		[parameter(ValueFromPipeline)]
+		[Alias("ServerInstance", "SqlServer", "SqlInstance")]
+		[DbaInstanceParameter[]]$ComputerName = $env:COMPUTERNAME,
+		[System.Management.Automation.PSCredential]$Credential,
+		[string]$CaServer,
+		[string]$CaName,
+		[string]$InstanceClusterName,
+		[securestring]$Password,
+		[string]$FriendlyName = "SQL Server",
+		[string]$CertificateTemplate = "WebServer",
+		[int]$KeyLength = 1024,
+		[switch]$Silent
+	)
+	begin {
+		
+		Test-RunAsAdmin
+		
+		if (!$CaServer -or !$CaName) {
+			try {
+				Write-Message -Level Output -Message "No CaServer or CaName specified. Finding it."
+				# hat tip Vadims Podans
+				$domain = ([System.DirectoryServices.ActiveDirectory.Domain]::GetCurrentDomain()).Name
+				$domain = "DC=" + $domain -replace '\.', ", DC="
+				$pks = [ADSI]"LDAP://CN=Enrollment Services, CN=Public Key Services, CN=Services, CN=Configuration, $domain"
+				$cas = $pks.psBase.Children
+				
+				$allcas = @()
+				foreach ($ca in $cas) {
+					$allcas += [pscustomobject]@{
+						CA = $ca | ForEach-Object { $_.Name }
+						Computer = $ca | ForEach-Object { $_.DNSHostName }
+					}
+				}
+			}
+			catch {
+				Stop-Function -Message "Cannot access Active Directory or find the Certificate Authority: $_"
+				return
+			}
+		}
+		
+		if (!$CaServer) {
+			$CaServer = ($allcas | Select-Object -First 1).Computer
+			Write-Message -Level Output -Message "Root Server: $CaServer"
+		}
+		
+		if (!$CaName) {
+			$CaName = ($allcas | Select-Object -First 1).CA
+			Write-Message -Level Output -Message "Root CA name: $CaName"
+		}
+		
+		$tempdir = ([System.IO.Path]::GetTempPath()).TrimEnd("\")
+		$certTemplate = "CertificateTemplate:$CertificateTemplate"
+		
+	}
+	
+	process {
+		
+		if (Test-FunctionInterrupt) { return }
+		
+		foreach ($computer in $computername) {
+			if (!$secondarynode) {
+				if (![dbavalidate]::IsLocalhost($computer) -and !$Password) {
+					Write-Message -Level Output -Message "You have specified a remote computer. A password is required for private key encryption/decryption for import."
+					$Password = Read-Host -AsSecureString -Prompt "Password"
+				}
+				
+				if ($InstanceClusterName) {
```

Comment:
Yes, it's a Windows cluster (wincluster in my AD), but it's also a SQL Instance Cluster (sqlcluster in my AD) - the SQL cluster name is what needs to be encrypted.